AI/src/main/java/dk/gruppeseks/bodtrd/ai/AIProvider.java

```diff
@@ -20,57 +22,48 @@
 public class AIProvider implements AISPI
 {
     private boolean[][] _grid;
+    private boolean[][] _explored;
 
     @Override
-    public Position[] getPath(Position start, Position goal, World world)
+    public Stack<Position> getPath(Position start, Position goal, World world) throws NoPathException
     {
-        //The following is dummy-code, untill the map is fixed
-        boolean[][] testGrid =
-        {
-            {
-                true, true, true, true
-            },
-            {
-                true, false, true, false
-            },
-            {
-                true, false, true, true
-            },
-            {
-                true, true, false, true
-            }
-        };
-        //End of dummy-code
-
         _grid = world.getMap().getGrid();
+        _explored = new boolean[_grid.length][_grid[0].length];
         int cellSize = world.getMap().getGridCellSize();
 
         int startX = (int)(start.getX() / cellSize);
         int startY = (int)(start.getY() / cellSize);
 
         int goalX = (int)(goal.getX() / cellSize);
         int goalY = (int)(goal.getY() / cellSize);
+        if (startX >= _grid.length || goalX >= _grid.length || startY >= _grid[0].length || goalY >= _grid[0].length)
+        {
+            throw new NoPathException("Bad start or end point");
+        }
+        if (startX < 0 || goalX < 0 || startY < 0 || goalY < 0)
+        {
+            throw new NoPathException("Bad start or end point");
+        }
 
         List<Node> path = AStarSearch(startX, startY, goalX, goalY);
-        Position[] output = new Position[path.size()];
+        Stack<Position> output = new Stack();
```

Comment:
was this done out of performance speculations?
pushing elements to a stack isnt more efficient than setting elements in an array for a fixed size, in fact it may be more expensive